Step2

wooki0108 opened the pull request:
test branch step2

wooki0108 committed:
add : 2회차 수업 실습 예제

wooki0108 committed:
Changes by bfine

wooki0108 committed:
test : branch step2

wooki0108 committed:
test : branch step2

wooki0108 committed:
Q1 - Rest API 특징

wooki0108 committed:
add : motel 등록, 조회 작성

bfine-lesson commented on src/main/java/com/lesson/spring/motel/api/dto/MotelCreateRequest.java:
```diff
@@ -0,0 +1,12 @@
+
+    private String name;
+    private int price;
+
```
보통 client request 나 server response 관련 dto 들은 
불변이이서 final 붙이는건 습관처럼 써주세요!

bfine-lesson commented on src/main/java/com/lesson/spring/motel/dao/MotelDao.java:
```diff
@@ -0,0 +1,10 @@
+public interface MotelDao {
+
+    List<MotelResponse> motelList();
+    void saveMotel(int price, String name);
```
Quiz. interface에 메서드 선언시 생략되는 부분이 무엇이 있을까요?

wooki0108 replied:
필드 : public static final
메소드는 : public abstract
생략 가능합니다.

bfine-lesson commented on src/main/resources/application.yml:
```diff
@@ -9,3 +9,13 @@ spring:
+      path: /h2-console
+  jpa:
+    hibernate:
+      ddl-auto=create:
```
혹시 create 때문에 JPA 추가 하신 건가요?! 
 보통은 = 대신 : 쓰고 있어요
  ddl-auto : create

wooki0108 replied:
네 h2 db에 넣어서 값 확인해 보려고 추가했습니다.
넵 수정했습니다 !

bfine-lesson commented on src/main/java/com/lesson/spring/motel/api/MotelApi.java:
```diff
@@ -0,0 +1,41 @@
+
+    // 모텔 등록
+    @PostMapping("/motel")
+    public void saveMotel(@RequestBody MotelCreateRequest request) {
```
/motel로 하면 endpoint가 2개가 되어서 관리하기 불편하지 않을까요?!

wooki0108 replied:
제가 이해를 잘 못했는데
예를 들면 @GetMapping("/motel"), @PostMapping("/motel") 이럴 때 endpoint 2개가 되어서 관리하기가 불편할 거 같다는 말씀이신가요?

bfine-lesson replied:
엇 아뇨  @GetMapping("/motels"), @PostMapping("/motel") 요렇게 되어있어서 질문드렸습니다~

bfine-lesson commented on src/main/java/com/lesson/spring/motel/api/MotelApi.java:
```diff
@@ -0,0 +1,41 @@
+    // 모텔 전체 조회
+    @GetMapping("/motels")
+    public List<MotelResponse> motelList() {
+        return motelService.getMotelList();
```
Quiz. MotelResponse name을 ownerName으로 바꾸고 싶다는 요청이 오면 어떻게 하실건가요?!

bfine-lesson commented on src/main/java/com/lesson/spring/motel/api/dto/MotelResponse.java:
```diff
@@ -0,0 +1,14 @@
+
+//    private Long id;
+    private String name;
+    private int price;
```
보통 가격에는 금액이 넘지 않아도 보통 long을 써요!

wooki0108 commented:
리뷰 감사합니당! 나중에 다 수정하겠습니다.
motel로 먼저 한번 만들어본 다음에 hotel 과제 진행하겠습니다.

wooki0108 committed:
Q2 - REST API 의 단점

wooki0108 committed:
refactor : dto final 키워드 추가
궁금한 게 있습니다.
커밋 메시지 컨벤션보면
fix : 버그 수정,
refactor : 프로덕션 코드 리팩토링
이런식으로 나오는데
여기서는 refactor 를 커밋 메시지로 사용하는 건가요?

wooki0108 committed:
refactor : ddl-auto=create -> ddl-auto : create

wooki0108 committed:
refactor : int price -> long price 로 변경

wooki0108 committed:
Q3 - @RequestBody vs @ModelAttribute

wooki0108 committed:
Q4 - @Controller vs @RestController 차이와 @RestController 사용했을때 Spring MVC 설정의 변화

wooki0108 committed:
Q5 - DAO vs Repository 차이

wooki0108 committed:
Q6 - Build Tool

bfine-lesson commented on README.md:
```diff
@@ -37,20 +37,26 @@
+ @RestController 사용 시 Spring MVC 설정 변화
+    - @RestController 사용 시 객체를 반환하기 때문에 HttpStatus를 설정할 수 없음 → 객체를 상황에 맞는 ResponseEntity로 감싸서 반환해야 함
+ 참고 : https://mangkyu.tistory.com/49
 ~~~
```
MVC 변화가 빠진것 같아요!

bfine-lesson commented on README.md:
```diff
@@ -3,52 +3,96 @@
-
+ 1. HTTP 프로토콜에 의존 → 메소드 형태가 제한적
+ 2. 표준의 부재
+    -> 관리의 어려움과 좋은(공식화 된) API 디자인 가이드가 존재하지 않음
```
이것 보다는 두가지 문제가 있어요!

wooki0108 committed:
step2
